### PR TITLE
GBM prediction code

### PR DESCRIPTION
**Pull request by davharris**
This code loops through the species and uses GBM to make predictions about the (Poisson-distributed) abundances.  It would take a while with 8 cores, so we should probably use Hipergator.

### CODE REVIEW

File: munge/train-test.R [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  scaled_x = scale(x)
+  
+  # Half the variance in X should come from temp and half from precip,
+  # but 3/4 of the columns are about temperature.
+  temp_columns = grep("^t", colnames(scaled_x))
+  scaled_x[ , temp_columns] = scaled_x[ , temp_columns] / sqrt(3)
+  
+  ica = fastICA(scaled_x, 10)
+  
+  function(new_x){
```
Comment (ethanwhite) [inline munge/train-test.R]: This must be something I don't understand about R yet. How does this nested function end up getting executed?

**Comment by ethanwhite** [comment]
Looks great overall.

The one main thing is my confusion as to how the execution of the unnamed function happens. Assuming this is basically a lambda function of some kind, then for something this length I think it's probably easier to understand as it's own separate (non-nested) named function, but since I don't really follow how it's being executed this may be something you have to do to get things to work.

The only other thing that I noticed is that I think a lot of the data management code could be made easier to read using `dplyr`, but that may just be a personal preference. E.g.,

```
train_climate = select(train_bbs, matches("ppt|tmin|tmean|tmax"))
train_rescaler = make_rescaler(train_climate)
```
or even

```
train_rescaler = make_rescaler(select(train_bbs, matches("ppt|tmin|tmean|tmax"))
```

instead of

```
climate_colnames = grep("ppt|tmin|tmean|tmax", colnames(train_bbs), value = TRUE)
train_rescaler = make_rescaler(train_bbs[ , climate_colnames])
```

**Comment by ethanwhite** [comment]
hey @davharris - just wanted to see if you might have the chance to finish the recommended changes so that we can get this merged in. I'm also happy to go ahead and merge this one as is if you don't want to make any changes.

**Comment by davharris** [comment]
Ack.  When did this merge happen? I think someone needs to take the command line away from me whenever I get sick.

Sorry!

**Comment by davharris** [comment]
Oh no.  I set weecology/bbs-forecasting as my `upstream` remote *and* as my `origin` remote. So when I tried to push to my own master branch, I ended up pushing here.  Fixing now...

Really sorry!

**Comment by ethanwhite** [comment]
No worries. Just go ahead and submit a new PR with any changes you want to make and reference this one so that it's easy to go back and look at the conversation.

**Pull request merged**